### PR TITLE
328 - Display of multiple modalities in sample table needs separator

### PR DESCRIPTION
**Pull request by nozomione**
## Issue Number
#328 

**API:** https://api.scpca.alexslemonade.org/v1/projects/SCPCP000009

## Purpose/Implementation Notes

Added a separator for multiple modalities in the sample table.

## Types of changes

What types of changes does your code introduce?

- Bugfix (non-breaking change which fixes an issue)

## Functional tests

List out the functional tests you've completed to verify your changes work locally.

## Checklist

- [x] Lint and unit tests pass locally with my changes

### CODE REVIEW

File: client/src/components/ProjectSamplesTable.js [hunk 1]
```diff
@@ -78,7 +78,12 @@ export const ProjectSamplesTable = ({
     { Header: 'Technology', accessor: 'technologies' },
     {
       Header: 'Other Modalities',
-      accessor: ({ modalities }) => modalities || 'N/A'
+      accessor: ({ modalities }) =>
+        modalities.map((modality, i) => (
+          <>
+            {i ? ' ,' : ''} {modality}
+          </>
+        )) || 'N/A'
```
Comment (jashapiro) [inline client/src/components/ProjectSamplesTable.js:86]: Thanks for doing this so fast!

As I'm not really sure of the data structures here, I'm not sure whether we can just use `modalities.join(', ')` to do this more simply as we do for the additional metadata? Are the `<>` & `</>` required?

Either way, we should reverse the comma space order, as suggested below:
```suggestion
      accessor: ({ modalities }) =>
        modalities.map((modality, i) => (
          <>
            {i ? ',' : ''} {modality}
          </>
        )) || 'N/A'
```

Comment (nozomione) [reply client/src/components/ProjectSamplesTable.js]: Revised the spacing (oops), thank you Josh!

**Note:** The `<></>`(short form of `<React.Fragment>`) is called [fragments](https://reactjs.org/docs/fragments.html) and is the way to properly render [JSX](https://reactjs.org/docs/introducing-jsx.html) in React UI :)

**Review by jashapiro — APPROVED** [review]
Thanks again for getting to this so fast! 

I have one more question, which is not about a change here, as this seems to preserve existing behavior, but about what might come after this in code. When I look at the portal now, if there are no "other modalities" for a sample, that cell is blank, which seems correct. But here there is an `N/A` being filled in: does that get replaced later with a blank? Or is this a change that hasn't yet percolated to the live version of the portal, and we will display "N/A" in the future?

**Comment by dvenprasad** [comment]
Based on the designs, I would expect blanks to be populated as `NA`. 
@jashapiro  Is a blank cell preferred? We can also use `None` which might work better for this field. 

Screen Shot 2022-11-28 at 2 33 26 PM

**Review by davidsmejia — APPROVED** [review]
LGTM

File: client/src/components/ProjectSamplesTable.js [hunk 2]
```diff
@@ -78,7 +78,9 @@ export const ProjectSamplesTable = ({
     { Header: 'Technology', accessor: 'technologies' },
     {
       Header: 'Other Modalities',
-      accessor: ({ modalities }) => modalities || 'N/A'
+      accessor: ({ modalities }) => (
+        <>{modalities.length ? <>{modalities.join(', ')}</> : 'N/A'}</>
```
Comment (davidsmejia) [inline client/src/components/ProjectSamplesTable.js:82]:
```suggestion
        <>{modalities.length ? modalities.join(', '): 'N/A'}</>
```

doesn't this work all the same?

**Comment by jashapiro** [comment]
> Based on the designs, I would expect blanks to be populated as `NA`. @jashapiro Is a blank cell preferred? We can also use `None` which might work better for this field.
> 
> Screen Shot 2022-11-28 at 2 33 26 PM

I think "None" is better for this field (and maybe also for "Additional Metadata Fields", though I expect that is never null).

File: client/src/components/ProjectSamplesTable.js (hunk 2 above)
Comment (nozomione) [reply client/src/components/ProjectSamplesTable.js]: Complented, thank you!

**Pull request merged**